pandas/io/stata.py

```diff
@@ -467,9 +464,9 @@ def g(x: datetime) -> int:
     if bad_loc.any():
         dates = Series(dates)
```

Comment:
```suggestion
```

And then below could use the existing `dates[bad_loc]` ? (if we are otherwise just setting on the `_values`, there is not much point in converting to a Series?)